src/get-or-create-render-context.js

```diff
@@ -7,53 +7,53 @@
 const vm = require("vm");
 // TODO(csilvers): try to get rid of the dependency on jsdom
 const jsdom = require("jsdom");
-const raf = require("raf");
 const cache = require("./cache.js");
 const profile = require("./profile.js");
 
-const runInContext = function(context, fn) {
-    return vm.runInContext("(" + fn.toString() + ")()", context);
-};
+class CustomResourceLoader extends jsdom.ResourceLoader {
+    fetch(url, options) {
+        // eslint-disable-next-line no-console
+        console.log("FETCHING: " + url);
+        return super.fetch(url, options);
+    }
+}
 
-const createRenderContext = function(jsPackages, pathToClientEntryPoint) {
-    const sandbox = {};
+const runInContext = function(jsdomContext, fnOrText) {
+    const script = typeof fnOrText === "function"
+        ? "(" + fnOrText.toString() + ")()"
+        : fnOrText;
+
+    if (typeof script !== "string") {
+        throw new Error("Must be a function or text");
+    }
+    return jsdomContext.runVMScript(new vm.Script(script));
+};
 
+const createRenderContext = function(locationUrl, jsPackages, pathToClientEntryPoint) {
     // A minimal document, for parts of our code that assume there's a DOM.
-    const doc = jsdom.jsdom(
+    const context = new jsdom.JSDOM(
         "<!DOCTYPE html><html><head></head><body></body></html>", {
-            // Forward console logs from inside the VM to the real console
-            virtualConsole: jsdom.createVirtualConsole().sendTo(console),
-            // CKEditor, and maybe other things, check for
-            // location.href, and expect it to be non-empty.
-            url:  "http://www.khanacademy.org",
+            url:  locationUrl,
```

Comment:
We cannot mess with `window.location` how we used to; instead this is where we set the base url for the "page".